ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/LedgerStateReaderWithFingerprintsFromValues.scala

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.validator.preexecution
+import com.daml.ledger.participant.state.kvutils.Fingerprint
+import com.daml.ledger.validator.LedgerStateOperations
+import com.daml.ledger.validator.LedgerStateOperations.{Key, Value}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * An in-transaction ledger state reader that also returns value fingerprints based on their value.
```

Comment:
What do you mean by 'in-transaction' here?

Reply:
Removed as it's implied by using the state operations.